Packs/AggregatedScripts/Scripts/ClearUserSession/ClearUserSession.py

```diff
@@ -301,41 +319,214 @@ def create_readable_output(outputs: list):
+        str: The error message from the entry.
+
+    Raises:
+        ValueError: If no error is detected in the entry.
+    """
+    if not is_error_enhanced(entry):
+        # If no error is detected, raise the original ValueError
+        raise ValueError("execute_command result has no error entry. before using get_error_enhanced use is_error_enhanced")
+
+    content_lower = str(_get_content_lower(entry))  # if content lower was None, is_error_enhanced would have been false
+    # 1. Check for Not Found errors first, as they are very specific
+    if is_not_found_error(content_lower):
+        return "User not found."
+
+    # 2. Check for Authentication/Authorization errors next
+    if is_auth_authz_error(content_lower):
+        return "Authentication failed."
+
+    # 3. Resolve to general error
+    content = entry.get("Contents") or entry.get("Content")
+    return f"Unknown error occurred: {content}"
+
+
+def run_command(
+    user_id: str,
+    results_for_verbose: list[CommandResults],
+    clear_session_results: list[tuple[str, str, str]],
+    brand: str,
+    user_name: Optional[str] = None,
+):
+    clear_user_sessions_command = Command(
+        name=COMMANDS_BY_BRAND[brand],
+        args={ARG_NAME_BY_BRAND[brand]: user_id},
+        brand=brand,
+    )
+    if clear_user_sessions_command.is_valid_args():
+        readable_outputs, _, error_message = clear_user_sessions(clear_user_sessions_command)
+        results_for_verbose.extend(readable_outputs)
+        if not error_message:
+            clear_session_results.append((brand, "Success", f"User session was cleared for {user_name or user_id}"))
+        else:
+            failed_message = f"{brand}: {error_message.lstrip('#').strip()}"
+            demisto.debug(
+                f"Failed to clear sessions for {brand} user with ID {user_id}. "
+                f"Error message: {error_message}. Response details: {readable_outputs}."
+            )
+            clear_session_results.append((brand, "Failed", failed_message))
+
+
 """ MAIN FUNCTION """
 
 
 def main():
     try:
         args = demisto.args()
         users_names = argToList(args.get("user_name", ""))
+        user_ids_arg = argToList(args.get("user_id", ""))
         verbose = argToBoolean(args.get("verbose", False))
         brands = argToList(args.get("brands", DEFAULT_BRANDS))
 
-        outputs: list = []
         results_for_verbose: list[CommandResults] = []
-
         filtered_users_names, outputs = remove_system_user(users_names, brands)
 
+        for user_id in user_ids_arg:
+            clear_session_results: list[tuple[str, str, str]] = []
+            for brand in brands:
+                run_command(user_id, results_for_verbose, clear_session_results, brand)
+            for brand, result, message in clear_session_results:
+                user_output = {"Message": message, "Result": result, "Brand": brand, "UserId": user_id, "UserName": ""}
+                outputs.append(user_output)
```

Comment:
That is a really weird section.
1. Why not mapping first `user_name` -> `user_id` and than do this for all of the `set(user_id)`?
2. Why create a complicated dict `clear_session_results` instead of just returning the values in the run_command?